harness/nbjunit/src/org/netbeans/junit/NbTestCase.java

```diff
@@ -474,6 +476,8 @@ public synchronized void waitFinished(final int timeout) throws Throwable {
                     long now = System.nanoTime();
                     try {
                         runTest();
+                    } catch (AssumptionViolatedException ex) {
+                        // ignore, the test is assumed to be meaningless.
```

Comment:
awesome. This will make the test case show up as green though, not skipped, right?

I actually wanted to implement this myself but got distracted before I figured out how to communicate this to junit properly.

Reply:
Sadly - I didn't come with a way how to mark it skipped, here in JUnit3-style code. The actual `TestResutl` that came around at runtime only allowed to mark the test passed/failed/error. I am not particularly happy with marking skipped as passing but better than failure. 

An option would be to rewrite all those `assume` statements to if-returns.

Please suggest / elaborate.